### PR TITLE
[PARTI-202] Catalog Ingestor: Enable ingestions to partially update an item

### PR DESCRIPTION
**Pull request by guiquintelas**
# PR Description

This PR adds the following changes:

-  Add the `PATCH` ingestion type
- Updates documentation accordingly
- Adjust the `CatalogIngestionPayload` type to reflect the different required catalog fields between `PATCH` vs the other two

## ⚠️ Dependencies

- None

## 🕹 Demo

Performing a `PATCH` ingestion:

| Before | After |
| -- | -- | 
| image | image |

# How to test

## 🧪 What should be tested?

As a reviewer, you should test:

- Check items being updated
- Check groups being updated
- Check variations being updated
- Make sure only the desired fields are being updated
- Make sure the `ingest`  return type is working as expected
  - when type == `PATCH` => only id is required
  - when other => multiple fields must be required

## 📝 How to test?

Reproduce the following steps to test:

1. Replace the src/index.ts file content with:
```ts
import { CatalogIngestor } from "./catalogIngestor/index";
import { CatalogIngestionType } from "./catalogIngestor/types";

export * from "./catalogIngestor/types";
export * from "./catalogIngestor/index";

const catalogIngestor = new CatalogIngestor({
  apiToken: "<token>",
  apiKey: "<key>",
});

async function test() {
  await catalogIngestor.ingest(async () => {
    return {
      type: CatalogIngestionType.PATCH,
      data: {
        groups: [],
        items: [
          {
            id: "dev-tshirt",
            item_name: "Developer t-shirt - After Patch",
          },
        ],
        variations: [],
      },
    };
  });
}

// eslint-disable-next-line @typescript-eslint/no-floating-promises
test();
```
2. Modify it with your testing data.
3. Check `items` results directly in the customer dashboard
4. Check `groups` and `variations` using the node client, calling the `retreiveVariations` and `getItemGroups` functions

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -72,7 +72,7 @@ Check out our [API docs](https://docs.constructor.io/rest_api/full_catalog/#repl
```
Comment (pedro-lb) [inline README.md:72]: **Suggestion:** I think we need to run `npm run generate-docs` before merging this.

Comment (pedro-lb) [inline README.md:72]: **Suggestion:** Can we bump the version on `package.json` to prep for a new release?

**Review by pedro-lb — CHANGES_REQUESTED** [review]
Awesome work here @guiquintelas!

A few minor things, we should be good to go after solving those.

File: README.md (hunk 1 above)
Comment (guiquintelas) [reply README.md:72]: oohh  forgot about that
done bce2b9389ffdd1e3c5e3f546e314a8891237c550 ✅

Comment (guiquintelas) [reply README.md:72]: Yeah we can 👌 
29e68c855a60e0f76d690daad5b9819b6eeb03a8

**Review by pedro-lb — APPROVED** [review]
LGTM! Awesome work with the types and general structure here @guiquintelas! 🔥 

After testing this, I've confirmed it also works for adding or updating facets:

image

We can't remove facets with this current implementation (since we're not using the prefix), but that's completely okay for now.

We're good to go here! - Don't forget to create a new release after merging this, and we'll also publish it to npm.

**Pull request merged**